content/en/platform/corda/5.2/deploying-operating/vnodes/upgrade-cpi.md

```diff
@@ -24,7 +24,7 @@ To upgrade a CPI, do the following:
 
 1. [Set the state of the virtual node to MAINTENANCE]({{< relref "./state.md">}}).
 2. Ensure no flows are running (that is, all flows have either “COMPLETED”, “FAILED” or “KILLED” status). You can check the list of running flows using <a href ="../../reference/rest-api/openapi.html#tag/Flow-Management-API/operation/get_flow__holdingidentityshorthash_">`GET /api/v5_2/flow/<holdingidentityshorthash>` </a>.
-3. Send the checksum of the CPI to upgrade to using the PUT method of the <a href ="../../reference/rest-api/openapi.html#tag/Virtual-Node-API/operation/put_virtualnode__virtualnodeshortid__cpi__targetcpifilechecksum_">`api/v1/virtualnode/<virtualnodeshortid>/cpi/<targetcpifilechecksum>` endpoint</a>:
+3. If you are using the standard Corda-managed virtual note databases, send the checksum of the CPI to upgrade to using the PUT method of the <a href ="../../reference/rest-api/openapi.html#tag/Virtual-Node-API/operation/put_virtualnode__virtualnodeshortid__cpi__targetcpifilechecksum_">`api/v5_2/virtualnode/<virtualnodeshortid>/cpi/<targetcpifilechecksum>` endpoint</a>:
```

Comment:
This step still needs to happen but after the new step documented below

Reply:
Edited and re-ordered steps.